src/app/ApplicationTemplate.Shared/Business/Impl/ChuckNorrisService.cs

```diff
@@ -6,10 +6,12 @@
 using System.Threading.Tasks;
 using ApplicationTemplate.Client;
 using DynamicData;
+using Microsoft.Extensions.DependencyInjection;
 using Uno.Extensions;
 
 namespace ApplicationTemplate.Business
 {
+	[RegisterService(ServiceLifetime.Singleton)]
```

Comment:
How does it know which interface to use as the key?

Reply:
The underlying registration process uses Scrutor and its method AsImplementedInterfaces(), which register all interfaces. If your class implements 3 different interfaces, you will then have 3 registrations.